pkg/logs/config/config.go

```diff
@@ -176,51 +176,86 @@ func buildTCPEndpoints() (*Endpoints, error) {
 	return NewEndpoints(main, additionals, useProto, false, 0), nil
 }
 
+// LogsConfigKeys stores logs configuration keys stored in YAML configuration files
+type LogsConfigKeys struct {
+	UseCompression          string
+	CompressionLevel        string
+	ConnectionResetInterval string
+	LogsDDURL               string
+	LogsNoSSL               string
+	DDURL                   string
+	DevModeNoSSL            string
+	AdditionalEndpoints     string
+	BatchWait               string
+}
+
+// logsConfigDefaultKeys defines the default YAML keys used to retrieve logs configuration
+var logsConfigDefaultKeys = LogsConfigKeys{
+	UseCompression:          "logs_config.use_compression",
+	CompressionLevel:        "logs_config.compression_level",
+	ConnectionResetInterval: "logs_config.connection_reset_interval",
+	LogsDDURL:               "logs_config.logs_dd_url",
+	LogsNoSSL:               "logs_config.logs_no_ssl",
+	DDURL:                   "logs_config.dd_url",
+	DevModeNoSSL:            "logs_config.dev_mode_no_ssl",
+	AdditionalEndpoints:     "logs_config.additional_endpoints",
+	BatchWait:               "logs_config.batch_wait",
+}
+
 // BuildHTTPEndpoints returns the HTTP endpoints to send logs to.
 func BuildHTTPEndpoints() (*Endpoints, error) {
+	return BuildHTTPEndpointsWithConfig(logsConfigDefaultKeys, httpEndpointPrefix)
+}
+
+// BuildHTTPEndpointsWithConfig uses two arguments that instructs it how to access configuration parameters, then returns the HTTP endpoints to send logs to.
+func BuildHTTPEndpointsWithConfig(logsConfig LogsConfigKeys, endpointPrefix string) (*Endpoints, error) {
```

Comment:
I could be good to add more explantation about how this function is able to default to the "classic" log configuration, I took me some time to understand how it works.

Also it can be good to write unit-test to valid this logic. It should have been helpful to have unit-tests to understand how it handle backward compatibility.

Reply:
I added a sentence to describe how this function can replicate the legacy behavior.